smart-home-application/Device.cpp

```diff
@@ -0,0 +1,25 @@
+#include "Device.h"
+
+#include <vector>
+
+Device::Device(std::string device_name, bool status) : device_name(device_name), status(status) {};
+
+void Device::setDeviceName(std::string device_name){
+    this->device_name = device_name;
+};
+
+std::string Device::getDeviceName(){
+    return this->device_name;
+};
+
+void Device::setStatus(bool status) {
+    this->status = status;
+};
+
+bool Device::getStatus() {
+    return this->status;
+};
+
+std::vector<Sensor> Device::getSensors() {
```

Comment:
here you're returning a copy of the sensors that are contained by your device. Read about returning references vs values (a bit similar to passing them as arguments) and how that affects your logic.

Long story short, you have a silent bug in your code waiting to trigger